### PR TITLE
explicit multiple selection mode for ajax dropdowns

### PR DESCRIPTION
**Pull request by btry**
Bad behavior of technician filter on dashboards : the filter allows multiple selection but should allow selection of one item only.

I think there is currently only one multiple selection ajax dropdown (the new abstract rights dropdown made by @AdrienClairembault ), but I may be wrong. 

image

### CODE REVIEW

File: inc/abstractrightsdropdown.class.php [hunk 1]
```diff
@@ -79,6 +79,7 @@ public static function show(string $name, array $values): string {
       $params = [
          'values' => $values,
          'valuesnames' => self::getValueNames($values),
+         'multiple'    => true,
```
Comment (btry) [inline inc/abstractrightsdropdown.class.php:82]: I think this is the only JS Ajax Dropdown with multiple selection. Confirmation needed.

File: inc/html.class.php [hunk 2]
```diff
@@ -4643,25 +4643,28 @@ static function jsAdaptDropdown($id, $params = []) {
...
+         'multiple'            => false,
+         'values'              => [],
+         'valuesnames'         => [],
+         'on_change'           => '',
+         'width'               => '80%',
+         'placeholder'         => '',
+         'display_emptychoice' => false,
+         'specific_tags'       => [],
+      ];
+      $params = array_merge($default_options, $params);
```
Comment (btry) [inline inc/html.class.php:4658]: normalize method's parameters handling

File: inc/html.class.php [hunk 3]
```diff
@@ -4673,12 +4676,7 @@ static function jsAjaxDropdown($name, $field_id, $url, $params = []) {
       ];
 
       // manage multiple select (with multiple values)
-      if (isset($params['values'])
-         && (
-            count($params['values'])
-            || !isset($params['value'])
-         )
-      ) {
+      if ($params['multiple']) {
```
Comment (btry) [inline inc/html.class.php:4679]: use the parameter "multiple"  instead of guessing the silgne / multiple mode

File: inc/html.class.php [hunk 4]
```diff
@@ -4695,13 +4693,11 @@ static function jsAjaxDropdown($name, $field_id, $url, $params = []) {
       unset($params['value']);
       unset($params['valuename']);
 
-      if (!empty($params['specific_tags'])) {
```
Comment (btry) [inline inc/html.class.php:4698]: Normalization of parameters handling ensures there is always a "specific_tags" key; then the if statement becomes useless.

File: inc/netpoint.class.php [hunk 5]
```diff
@@ -128,10 +128,10 @@ static function dropdownNetpoint($myname, $value = 0, $locations_id = -1, $displ
...
       $field_id = Html::cleanId("dropdown_".$myname.$rand);
       $param    = ['value'               => $value,
-                        'valuename'           => $name,
-                        'entity_restrict'     => $entity_restrict,
-                        'devtype'             => $devtype,
-                        'locations_id'        => $locations_id];
+                   'valuename'           => $name,
+                   'entity_restrict'     => $entity_restrict,
+                   'devtype'             => $devtype,
+                   'locations_id'        => $locations_id];
```
Comment (btry) [inline inc/netpoint.class.php:134]: align the items

**Review by AdrienClairembault — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**